meter/homematic/connection.go

```diff
@@ -149,3 +145,26 @@ func (c *Connection) GridTotalEnergy() (float64, error) {
 	res, err := c.XmlCmd("getValue", c.MeterChannel, Param{CCUString: "IEC_ENERGY_COUNTER"})
 	return res.Value.CCUFloat, err
 }
+
+// ParseHmError checks on Homematic CCU error codes
+// Refer to page 30 of https://homematic-ip.com/sites/default/files/downloads/HM_XmlRpc_API.pdf
+func ParseHmError(res MethodResponse) error {
+	var faultCode int64
+	var faultString string
+
+	faultCode = 0
+	for i := 0; i < len(res.Fault); i++ {
+		if res.Fault[i].Name == "faultCode" {
+			faultCode = res.Fault[i].Value.CCUInt
+		}
+		if res.Fault[i].Name == "faultString" {
+			faultString = res.Fault[i].Value.CCUString
+		}
+	}
+
+	if faultCode != 0 {
+		return fmt.Errorf("CCU error: %s (faultcode %v)", faultString, faultCode)
```

Comment:
```suggestion
		return fmt.Errorf("%s (%v)", faultString, faultCode)
```

...dass es ein Fehler ist wissen wir ja schon.